asset-transfer-secured-agreement/application-gateway-typescript/src/app.ts

```diff
@@ -0,0 +1,434 @@
+        }
+    } catch (e) {
+        console.log(`${RED}*** Failed: VerifyAssetProperties - ${e}${RESET}`);
+    }
+}
+
+async function agreeToBuy(assetKey:string, org:string, price:number, contract:Contract):Promise<void> {
+    try {
+        const asset_price = {
+            asset_id: assetKey,
+            price:price,
+            trade_id: now.toString()
+        };
+        console.log(`${GREEN}--> Submit Transaction: AgreeToBuy, ${assetKey} as ${org} - endorsed by ${org}${RESET}`);
+
+        await contract.submit('AgreeToBuy',{
+            arguments:[assetKey],
+            transientData: { asset_price: JSON.stringify(asset_price) }
+        });
+
+        console.log(`*** Result: committed, ${org} has agreed to buy asset ${assetKey} for 100`);
+    } catch (e) {
+        console.log(`${RED}*** Failed: AgreeToBuy - ${e}${RESET}`);
+    }
+}
+
+async function readSalePrice(assetKey:string, org:string, contract:Contract):Promise<void> {
+    try {
+        console.log(`${GREEN}--> Evaluate Transaction: GetAssetSalesPrice, - ${assetKey} from organization ${org}${RESET}`);
+        if(org===mspIdOrg2){
+            console.log(`${GREEN}* Expected to fail as ${org} has not set a sale price${RESET}`)
+        }
+
+        const resultBytes = await contract.evaluateTransaction('GetAssetSalesPrice', assetKey);
+
+        const resultString = utf8Decoder.decode(resultBytes);
+        const result = JSON.parse(resultString);
+        console.log('*** Result: GetAssetSalesPrice', result);
+    } catch (e) {
+        console.log(`${RED}*** Failed evaluateTransaction GetAssetSalesPrice: ${e}${RESET}`);
+    }
+}
+
+async function readBidPrice(assetKey:string, org:string, contract:Contract):Promise<void> {
+    try{
+        console.log(`${GREEN}--> Evaluate Transaction: GetAssetBidPrice, - ${assetKey} from organization ${org}${RESET}`);
+        if(org===mspIdOrg1){
+            console.log(`${GREEN}* Expected to fail as Org1 has not agreed to buy${RESET}`)
+        }
+
+        const resultBytes = await contract.evaluateTransaction('GetAssetBidPrice', assetKey);
+
+        const resultString = utf8Decoder.decode(resultBytes);
+        const result = JSON.parse(resultString);
+
+        console.log('*** Result: GetAssetBidPrice', result);
+    } catch (e) {
+        console.log(`${RED}*** Failed evaluateTransaction GetAssetBidPrice: ${e}${RESET}`);
+    }
+}
+
+async function transferAsset(assetKey:string, org:string,buyerOrgID:string, price:number,contract:Contract):Promise<void> {
+    try{
+        const asset_properties = {
+            object_type: 'asset_properties',
+            asset_id: assetKey,
+            color: 'blue',
+            size: 35,
+            salt: randomBytes
+        };
+        const asset_price = {
+            asset_id: assetKey,
+            price:price,
+            trade_id: now.toString()
+        };
```

Comment:
Since asset properties and price get used multiple times throughout the file, it would be good practice to declare a module-scoped interface for **AssetProperties** and **AssetPrice**, and use them throughout the application so the compiler can help ensure your data is consistent.